src/bridge/AdminRoomHandler.ts

```diff
@@ -480,7 +482,7 @@ export class AdminRoomHandler {
                     `Username is longer than the maximum permitted by the bridge (${SANE_USERNAME_LENGTH}).`
                 );
             }
-            else if (IdentGenerator.sanitiseUsername(username) !== username) {
+            else if (IdentGenerator.sanitiseUsername(username) !== usernameWithoutUnderscores) {
```

Comment:
I think the best approach might be to make santitiseUsername more lax to be honest. The linked issue seems to mention that more characters are supported than I expected, so perhaps just make that accept more?